ui/packages/catalog-ui-search/src/main/webapp/component/result-item/result-item.collection.tsx

```diff
@@ -39,67 +52,109 @@ const ResultGroup = styled.div`
-                        replaceElement
-                      />
-                    )
-                  })}
-                </ResultItemCollection>
-              </div>
-            </ResultGroup>
-          )
-        } else {
-          return (
-            <MarionetteRegionContainer
-              key={result.id}
-              view={childView}
-              viewOptions={{
-                selectionInterface,
-                model: result,
-              }}
-              replaceElement
-            />
-          )
-        }
-      })}
+    <ResultItemCollection className={className}>
+      <div className="result-item-collection-empty">No Results Found</div>
     </ResultItemCollection>
   )
 }
 
+const getResultView = (props: Props) => {
+  const { results, selectionInterface, className, solrQuery } = props
+  const showingResultsFor = createShowResultText(solrQuery)
+
+  return (
+    <SolrQueryDisplay className={className}>
+      <div className="solr-query">{showingResultsFor}</div>
+      <ResultItemCollection
+        className={`${className} is-list has-list-highlighting`}
+      >
+        {results.map(result => {
+          if (result.duplicates) {
+            const amount = result.duplicates.length + 1
+            return (
+              <ResultGroup key={result.id}>
+                <div className="group-representation">{amount} duplicates</div>
+                <div className="group-results global-bracket is-left">
+                  <ResultItemCollection className="is-list has-list-highlighting">
+                    <MarionetteRegionContainer
+                      view={childView}
+                      viewOptions={{
+                        selectionInterface,
+                        model: result,
+                      }}
+                      replaceElement
+                    />
+                    {result.duplicates.map((duplicate: any) => {
+                      return (
+                        <MarionetteRegionContainer
+                          key={duplicate.id}
+                          view={childView}
+                          viewOptions={{
+                            selectionInterface,
+                            model: duplicate,
+                          }}
+                          replaceElement
+                        />
+                      )
+                    })}
+                  </ResultItemCollection>
+                </div>
+              </ResultGroup>
+            )
+          } else {
+            return (
+              <MarionetteRegionContainer
+                key={result.id}
+                view={childView}
+                viewOptions={{
+                  selectionInterface,
+                  model: result,
+                }}
+                replaceElement
+              />
+            )
+          }
+        })}
+      </ResultItemCollection>
+    </SolrQueryDisplay>
+  )
+}
+
+function createShowResultText(solrQuery: any[]) {
```

Comment:
❓ Can you provide some example input/output of this function? This area seems overly complicated to me. I tried cleaning it up but I think there are some bugs when it comes to appending the `... more` text.

Reply:
Examples are: 
`Showing Results for 'fast'`
`Showing Results for 'fast', 'normal'`
`Showing Results for 'fast', 'normal', ... 3 more` (the ... # more will occur when the string length reaches >69 characters). 
This means there are several cases to consider/not consider
`Showing Results for ` consists of 11 characters, which means we have 58 characters of bandwidth to hold search field values.

1. There is no word in the english language that has >58 characters so we are guaranteed that one of the fields will always show. In the case that a user types in 100 characters of garbage, the spellcheck won't even give a suggestion. Basically there's no chance of having a >58 character length string in the array. 
2. `... # more` takes 10-11 characters (single digit, or double digit more fields), so we need to check if the next field that was searched on will overtake the 58 character bandwidth. If so, we append the `... # more` instead of the field value.